asmcomp/arm64/proc.ml

```diff
@@ -254,23 +254,25 @@ let regs_are_volatile _rs = false
 
 (* Registers destroyed by operations *)
 
-let destroyed_at_c_call =
-  (* x19-x28, d8-d15 preserved *)
+let destroyed_at_c_noalloc_call =
+  (* x20-x28, d8-d15 preserved *)
   Array.of_list (List.map phys_reg
-    [0;1;2;3;4;5;6;7;8;9;10;11;12;13;14;15;
+    [0;1;2;3;4;5;6;7;8;9;10;11;12;13;14;15;16;
```

Comment:
With a minor change to arm64.S, we might be able to avoid destroying x19, in which case this change can be reverted.

Reply:
So using the frame pointer to avoid destroying `x19` is something I think can work by changing the emit for `Iextcall` in the case of non-allocating C calls. 

Let me try that.

Reply:
e7b3839 implements `Iextcall` for non-allocating C calls using the frame pointer register to hold the OCaml stack. 

If we end up having frame pointer support on the OCaml frames in arm64, then I believe we could leave `x29` alone and then restore the OCaml stack using `x29` and `env.stack_offset`. So I think this doesn't back us into a corner there. 

Overall I like this change; it gives back `x19` to user code.